IntegrationTests/IntegrationScenarios/Scenarios/Core/StopCoreScenarioTests.swift

```diff
@@ -0,0 +1,185 @@
+    /// * sends log and trace from home screen,
+    /// * goes to picture screen and downloads the image,
+    /// * goes back to the home screen.
+    private func playScenario(from root: CSRootScreen = CSRootScreen()) -> CSRootScreen {
+        let home = root.tapGoToHome()
+        home.tapTestLogging()
+        home.tapTestTracing()
+        let pictureScreen = home.tapTestRUM()
+        pictureScreen.tapDownloadImage()
+        pictureScreen.waitForImageBeingDownloaded()
+        return pictureScreen
+            .tapBack()
+            .tapBack()
+    }
+
+    // MARK: - Data assertions
+
+    private func assertLoggingDataWasCollected(by serverSession: ServerSession) throws {
+        let recordedRequests = try serverSession.pullRecordedRequests(timeout: dataDeliveryTimeout) { requests in
+            try LogMatcher.from(requests: requests).count == 1
+        }
+
+        assertLogging(requests: recordedRequests)
+
+        let logMatchers = try LogMatcher.from(requests: recordedRequests)
+        XCTAssertEqual(logMatchers.count, 1)
+        let logMatcher = logMatchers[0]
+        logMatcher.assertMessage(equals: "test message")
+        logMatcher.assertStatus(equals: "info")
+    }
+
+    private func assertTracingDataWasCollected(by serverSession: ServerSession) throws {
+        let recordedRequests = try serverSession.pullRecordedRequests(timeout: dataDeliveryTimeout) { requests in
+            try SpanMatcher.from(requests: requests).count == 1
+        }
+
+        assertTracing(requests: recordedRequests)
+
+        let spanMatchers = try SpanMatcher.from(requests: recordedRequests)
+        XCTAssertEqual(spanMatchers.count, 1)
+        let spanMatcher = spanMatchers[0]
+        XCTAssertEqual(try spanMatcher.operationName(), "test span")
+    }
+
+    private func assertRUMDataWasCollected(by serverSession: ServerSession) throws {
+        let recordedRequests = try serverSession.pullRecordedRequests(timeout: dataDeliveryTimeout) { requests in
+            // ignore telemetry and application start view events
+            let eventMatchers = try requests
+                .flatMap { request in try RUMEventMatcher.fromNewlineSeparatedJSONObjectsData(request.httpBody) }
+                .filterTelemetry()
+                .filterApplicationLaunchView()
+
+            guard let session = try RUMSessionMatcher.groupMatchersBySessions(eventMatchers).first else {
+                return false
+            }
+
+            return session.views.count >= 3
+        }
+
+        assertRUM(requests: recordedRequests)
+
+        // ignore telemetry and application start view events
+        let eventMatchers = try recordedRequests
+            .flatMap { request in try RUMEventMatcher.fromNewlineSeparatedJSONObjectsData(request.httpBody) }
+            .filterTelemetry()
+            .filterApplicationLaunchView()
+
+        let session = try XCTUnwrap(RUMSessionMatcher.groupMatchersBySessions(eventMatchers).first)
```

Comment:
It doesn't work sadly, because I run the scenario 3 times:
1. After first init
2. After stopping the core
3. After second init

The assertion aims to cover the first and last run: the first run will have the `application-launch/view` but not the 3rd. 
Also, I'm not triggering the end-session screens because I need to replay the scenario :/ 

I can still try to clean up by having 2 assertion methods for RUM.